### PR TITLE
Feat default response headers

### PR DESCRIPTION
**Pull request by lohanidamodar**

### CODE REVIEW

**Comment by lohanidamodar** [comment]
@eldadfux ready for review. The test was passing, I mistakenly closed and reopend.

**Comment by eldadfux** [comment]
@christyjacob4 @TorstenDittmann this is a pretty wide change, please review as well.

File: templates/dotnet/src/Appwrite/Client.cs.twig [hunk 1]
```diff
@@ -36,7 +36,12 @@ namespace {{ spec.title | caseUcfirst }}
             this.headers = new Dictionary<string, string>()
             {
                 { "content-type", "application/json" },
-                { "x-sdk-version", "{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{sdk.version}}" }
+                { "x-sdk-version", "{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{sdk.version}}" } {% if spec.global.defaultHeaders | length > 0 %},{% endif %}
+
+        {% for key,header in spec.global.defaultHeaders %}
+        {"{{key}}", "{{header}}"}{% if not loop.last %},{% endif %}
```
Comment (TorstenDittmann) [inline templates/dotnet/src/Appwrite/Client.cs.twig:42]: `{ "{{key}}", "{{header}}" }{% if not loop.last %},{% endif %}`

Add spaces to fit previous code 🙂

File: templates/java/src/main/java/io/appwrite/Client.java.twig [hunk 2]
```diff
@@ -33,8 +33,13 @@ public class Client {
         this.endPoint = endPoint;
         this.selfSigned = selfSigned;
         this.headers = new HashMap<>(Map.ofEntries(
-                entry("content-type", "application/json"),
-                entry("x-sdk-version", "{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{ sdk.version }}")
+            entry("content-type", "application/json"),
+            entry("x-sdk-version", "{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{ sdk.version }}") {% if spec.global.defaultHeaders | length > 0 %},{% endif %}
```
Comment (TorstenDittmann) [inline templates/java/src/main/java/io/appwrite/Client.java.twig:37]: `entry("x-sdk-version", "{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{ sdk.version }}"){% if spec.global.defaultHeaders | length > 0 %},{% endif %}`

remove the space before the to be added `,` 🙂

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]
Just some minor stuff like whitespace. Rest looks good 👍🏻

File: templates/dotnet/src/Appwrite/Client.cs.twig (hunk 1 above)
Comment (eldadfux) [reply templates/dotnet/src/Appwrite/Client.cs.twig:42]: +1

**Comment by lohanidamodar** [comment]
@TorstenDittmann please check the latest push, if it fixes all that is required. Thanks

**Review by TorstenDittmann — APPROVED** [review]
lgtm now 👍🏻 🙂

**Comment by eldadfux** [comment]
@lohanidamodar , @TorstenDittmann I still see a lot of potential indentations issues like this one:
image

**Comment by TorstenDittmann** [comment]
> @lohanidamodar , @TorstenDittmann I still see a lot of potential indentations issues like this one:
> image

I've run the example.php and checked the output, there everything looked fine.

**Comment by lohanidamodar** [comment]
> @lohanidamodar , @TorstenDittmann I still see a lot of potential indentations issues like this one:
> image

If i do correct indentation in the twig file the generated code will have quite unintended spacing, not sure what can be done. I had to move those codes with twigs expression towards left to adjust the indentation of the generated code. Let me know if I am doing something wrong.

**Comment by eldadfux** [comment]
image


@lohanidamodar The solution might be to remove all indentations from the loop statement and keep the code on as the above line, let me know if that works

**Comment by lohanidamodar** [comment]
Will try that

**Comment by lohanidamodar** [comment]
> image
> 
> @lohanidamodar The solution might be to remove all indentations from the loop statement and keep the code on as the above line, let me know if that works

@eldadfux Great suggestion, worked like a charm. Thanks!

**Review by TorstenDittmann — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]
LGTM. Awesome work @lohanidamodar  🥳

**Pull request merged**